Assets/Labrys/FeatureLibrary.cs

```diff
@@ -0,0 +1,195 @@
+            library.TargetDirectory = loadPath;
+            library.Refresh();
+
+            ProjectWindowUtil.CreateAsset(library, savePath);
+        }
+
+        public void Refresh()
+        {
+            string[] assetGUIDs = AssetDatabase.FindAssets($"t:{nameof(FeatureAsset)}", new string[] { TargetDirectory });
+            nameToFAID = new Dictionary<string, ushort>();
+            features = new FeatureAsset[assetGUIDs.Length];
+
+            foreach (string guid in assetGUIDs)
+            {
+                string path = AssetDatabase.GUIDToAssetPath(guid);
+                Add(path.Replace(TargetDirectory, "").Replace(".asset", "").TrimStart('/'),
+                    AssetDatabase.LoadAssetAtPath<FeatureAsset>(path));
+            }
+        }
+#endif
+        private FeatureLibrary()
+        {
+            nameToFAID = new Dictionary<string, ushort>();
+            features = new FeatureAsset[0];
+        }
+
+        private bool Add(string name, FeatureAsset feature)
+        {
+            if (name == null || feature == null)
+                return false;
+
+            features[nameToFAID.Count] = feature;
+            nameToFAID.Add(name, (ushort)nameToFAID.Count);
+
+            return true;
+        }
+
+        public FeatureAsset GetFeature(string name) => this[name];
+
+        public FeatureAsset GetFeature(ushort faid) => this[faid];
+
+        public bool Contains(string name) => nameToFAID.ContainsKey(name);
+
+        public bool Contains(ushort faid) => 0 <= faid && faid < features.Length;
+
+        public ushort GetId(string name)
+        {
+            if (nameToFAID.TryGetValue(name, out ushort faid))
+            {
+                return faid;
+            }
+            throw new ArgumentException($"Unknown name: {name}.");
+        }
+
+        public IEnumerator<Entry> GetEnumerator()
+        {
+            if (nameToFAID == null)
+                yield break;
+
+            foreach(KeyValuePair<string, ushort> entry in nameToFAID)
+            {
+                yield return new Entry(entry.Value, entry.Key, this[entry.Value]);
+            }
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        public object Clone()
+        {
+            FeatureLibrary clone = CreateInstance<FeatureLibrary>();
+
+            clone.TargetDirectory = TargetDirectory;
+
+            clone.features = new FeatureAsset[features.Length];
+            Array.Copy(features, clone.features, features.Length);
+
+            clone.nameToFAID = new Dictionary<string, ushort>();
+            foreach(KeyValuePair<string, ushort> pair in nameToFAID)
+            {
+                clone.nameToFAID.Add(pair.Key, pair.Value);
+            }
+
+            return clone;
+        }
+
+        public override string ToString()
+        {
+            string str = "";
+
+            foreach(Entry e in this)
```

Comment:
i'm guessing this is implicitly calling `IEnumerable.GetEnumerator()`- neat syntax